src/poetry/utils/env/virtual_env.py

```diff
@@ -133,3 +135,14 @@ def temp_environ(self) -> Iterator[None]:
 
     def _updated_path(self) -> str:
         return os.pathsep.join([str(self._bin_dir), os.environ.get("PATH", "")])
+
+    def includes_system_site_packages(self) -> bool:
+        pyvenv_cfg = self._path / "pyvenv.cfg"
+        return "include-system-site-packages = true" in pyvenv_cfg.read_text()
+
+    def is_path_relative_to_lib(self, path: Path) -> bool:
+        system_env = SystemEnv(Path(sys.prefix))
+        return (
+            self.includes_system_site_packages()
+            and system_env.is_path_relative_to_lib(path)
+        ) or super().is_path_relative_to_lib(path)
```

Comment:
```suggestion
        return super().is_path_relative_to_lib(path) or (
            self.includes_system_site_packages()
            and SystemEnv(Path(sys.prefix)).is_path_relative_to_lib(path)
        )
```

That way we don't have to create a SystemEnv and call additional methods in the default case (where system site packages are not included).